apt/minimization/minimizer.py

```diff
@@ -409,40 +486,114 @@ def transform(self, X: Optional[DATA_PANDAS_NUMPY_TYPE] = None, features_names:
         if not self._features:
             self._features = [i for i in range(x.shape[1])]
 
-        mapped = np.zeros(x.shape[0])  # to mark records we already mapped
-        all_indexes = []
-        for i in range(len(self.cells)):
-            indexes = self._get_record_indexes_for_cell(x, self.cells[i], mapped)
-            all_indexes.append(indexes)
-        generalized = self._generalize_indexes(x, self.cells, all_indexes)
+        if self._dt:  # only works if fit was called previously (but much more efficient)
+            nodes = self._get_nodes_level(self._level)
+            QI = x.loc[:, self.features_to_minimize]
+            used_x = x
+            if self.train_only_features_to_minimize:
+                used_x = QI
+            prepared = self._encode_categorical_features(used_x)
+            generalized = self._generalize(x, prepared, nodes, self.cells, self._cells_by_id)
+        else:
+            mapped = np.zeros(x.shape[0])  # to mark records we already mapped
+            all_indexes = []
+            for i in range(len(self.cells)):
+                indexes = self._get_record_indexes_for_cell(x, self.cells[i], mapped)
+                all_indexes.append(indexes)
+            generalized = self._generalize_indexes(x, self.cells, all_indexes)
 
         if dataset and dataset.is_pandas:
             return generalized
         elif isinstance(X, pd.DataFrame):
             return generalized
         return generalized.to_numpy()
 
+    def _calc_ncp_for_generalization(self, generalization, range_counts, category_counts, total_count):
+        total_ncp = 0
+        total_features = len(generalization['untouched'])
+        ranges = generalization['ranges']
+        categories = generalization['categories']
+
+        # suppressed features are already taken care of within _calc_ncp_numeric
+        for feature in ranges.keys():
+            feature_ncp = self._calc_ncp_numeric(ranges[feature], range_counts[feature],
+                                                 self._feature_data[feature], total_count)
+            total_ncp = total_ncp + feature_ncp
+            total_features += 1
+        for feature in categories.keys():
+            feature_ncp = self._calc_ncp_categorical(categories[feature], category_counts[feature],
+                                                     self._feature_data[feature],
+                                                     total_count)
+            total_ncp = total_ncp + feature_ncp
+            total_features += 1
+        if total_features == 0:
+            return 0
+        return total_ncp / total_features
+
+    @staticmethod
+    def _calc_ncp_categorical(categories, category_count, feature_data, total):
```

Comment:
Maybe feature_category_count**s** and **feature_**category_sizes for consistency?